desdeo_tools/utilities/model_management.py

```diff
@@ -0,0 +1,85 @@
+from numpy.core.numeric import indices
+from numpy.lib.arraysetops import unique
+from desdeo_tools.scalarization.ASF import SimpleASF
+from numba import njit
+import numpy as np
+from desdeo_problem.problem.Objective import ScalarDataObjective
+import pandas as pd
+import copy
+from desdeo_problem.surrogatemodels.SurrogateModels import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import DotProduct,\
+    WhiteKernel, RBF, Matern, ConstantKernel
+
+
+
+def remove_duplicate(X, archive_x):
+    """identifies of the duplicate rows for decision variables"""
+    indicies = None
+    archive = archive_x.to_numpy()
+    tmp = X
+    for i in archive:
+        for k in range(len(X)):
+            for j in range(len(i)):
+                tmp[k,j] = X[k,j] - i[j]
+        tmp = np.round(tmp, 3)
+        if indicies is None:
+            indicies = np.where(~tmp.any(axis=1))[0]
+        else:
+            tmp = np.where(~tmp.any(axis=1))[0]
+            if tmp.size > 0:
+                indicies = np.hstack((indicies.squeeze(),tmp))
+        if indicies.size == 0:
+            indicies = None
+    if indicies is None:
+        return None
+    else:
+
+        return indicies
+
+
+def ikrvea_mm(reference_point, evolver , problem, u: int) -> float:
```

Comment:
Do you need the evolver at all? You never use the EA methods. Instead, you only use the objective/decision/uncertainity vectors.